### PR TITLE
Fix viewer crash when trigger CAH in high temperature

### PR DESCRIPTION
**Pull request by Nir-Az**
Add exception catch on triggering CAH from the viewer.
Convert exception error to an error popup message.

### CODE REVIEW

File: common/cah-model.cpp [hunk 1]
```diff
@@ -93,7 +93,17 @@ bool cah_model::prompt_trigger_popup(ux_window& window, const std::string& error
                 {
                     auto sd = *itr;
                     global_calib_status = RS2_CALIBRATION_RETRY; // To indicate in progress state
-                    sd->s->set_option(RS2_OPTION_TRIGGER_CAMERA_ACCURACY_HEALTH, 1.0f);
+                    try
+                    {
+                        sd->s->set_option(RS2_OPTION_TRIGGER_CAMERA_ACCURACY_HEALTH, 1.0f);
```
Comment (maloel) [inline common/cah-model.cpp]: Please don't use 1.0f -- we have an enum now. Use RS2_CAH_TRIGGER_NOW

File: common/cah-model.cpp [hunk 2]
```diff
@@ -93,7 +93,17 @@ bool cah_model::prompt_trigger_popup(ux_window& window, const std::string& error
...
+                    }
+                    catch (std::exception const & e)
+                    {                        
+                        error_message = to_string() << "Trigger Calibration Failure:\n" << e.what();
```
Comment (maloel) [inline common/cah-model.cpp]: No need to capitalize every word :)

File: common/cah-model.cpp [hunk 3]
```diff
@@ -186,7 +196,14 @@ bool cah_model::prompt_reset_popup(ux_window& window, const std::string& error_m
...
                 auto sd = *itr;
                 // Trigger CAH process
-                sd->s->set_option(RS2_OPTION_RESET_CAMERA_ACCURACY_HEALTH, 1.0f);
+                try
+                {
+                    sd->s->set_option(RS2_OPTION_RESET_CAMERA_ACCURACY_HEALTH, 1.0f);
+                }
+                catch (std::exception const & e)
+                {
+                    error_message = to_string() << "Calibration Reset Failure:\n" << e.what();
```
Comment (maloel) [inline common/cah-model.cpp]: Capitalization

**Review by maloel — APPROVED** [review]

File: common/cah-model.cpp (hunk 1 above)
Comment (Nir-Az) [reply common/cah-model.cpp]: Done

File: common/cah-model.cpp (hunk 2 above)
Comment (Nir-Az) [reply common/cah-model.cpp]: Done

File: common/cah-model.cpp (hunk 3 above)
Comment (Nir-Az) [reply common/cah-model.cpp]: Done

**Pull request merged**